README.rst

```diff
@@ -0,0 +1,165 @@
+
+
+.. code-block:: bash
+
+    yarn global add elm-docstyle
+
+
+Detailed Instructions
+=====================
+
+``elm-docstyle [elm_code_directory | path_to_elm_file]``
+
+Options:
+    --help, -h          Print the help output.
+    --version, -v       Print the package version.
+    --verbose           If set, the offending comments are added to the error report in full.
+    --check_all         If set, all the function declarations are checked (including the non-exported ones).
+                        Otherwise, only the exported function declarations are checked.
+    --config_path       Path to the elm-docstyle JSON config. If unspecified, a default config is used.
+    --format            Output format ("human" or "json"). The default is "human".
+
+Elm-docstyle will look recursively for Elm files in all the directories given, excluding the ``elm-stuff``.
+
+
+Configuration
+=============
+
+You can specify a configuration as a .json file, which needs to contain the field "excludedChecks" as a list of
+strings. The excluded checks will be fed to the checker and ignored.
+
+Example of a elm-docstyle-config.json file:
+
+.. code-block:: JSON
+
+    {
+        "excludedChecks": ["TodoComment", "NoTopLevelComment"]
+    }
+
+
+
+Refer to the next section to know which checks are supported (and can hence be disabled).
+
+Supported Checks
+----------------
+
+==========================  ======================================================================================
+Name                        Explanation
+==========================  ======================================================================================
+``NotCapitalized``          the first word of the comment should be capitalized.
+``NoStartingSpace``         the comment should start with a space.
+``NoStartingVerb``          the comment should start with a verb in third person (stem -s).
+``NoEndingPeriod``          the first line of the comment should end with a period.
+``WrongCommentType``        the comment type "{-|-}" should not be used in a non-documentation comment.
+``TodoComment``             the comment should not contain the strings (with no dots) "t.o.d.o" or "f.i.x.m.e".
+``NoEntityComment``         a comment is expected on top of the declaration, but none was found.
+``NoTopLevelComment``       a comment is expected for the module, but none was found.
+``NotExistingArgument``     the name of a documented argument is not included in the declaration's argument names.
+``NotAnnotatedArgument``    the record field or function argument is not included in the documentation.
+==========================  ======================================================================================
+
+We follow the same convention as the Elm core libraries for including arguments in the documentation: specifying
+arguments in the format ``* `arg_name` -- explanation`` or ``* `arg_name` &mdash; explanation``, which render nicely in
+HTML. For instance:
```

Comment:
Please describe how the multi-line argument explanations should be handled (indented on the next line?).

Reply:
The ´elm-format´ forces indentation on the next line as:
```
  - ´checkAllDefinitions´ -- if set, all declarations are checked;
    otherwise, only exported ones.
```

Reply:
(also, notice the symbol "-" at the beginning, enforced by ´elm-format´. I'll change the rule, regex, comments and tests)